core/sr-primitives/src/lib.rs

```diff
@@ -258,6 +258,82 @@ impl From<codec::Compact<Perbill>> for Perbill {
 	}
 }
 
+/// Perquill is parts-per-quintillion. It stores a value between 0 and 1 in fixed point and
+/// provides a means to multiply some other value by that.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize, Debug))]
+#[derive(Encode, Decode, Default, Copy, Clone, PartialEq, Eq)]
+pub struct Perquill(u64);
+
+impl Perquill {
+	/// Returns the internal u64 value
+	pub fn extract(&self) -> u64 { self.0 }
+
+	/// Nothing.
+	pub fn zero() -> Perquill { Perquill(0) }
+
+	/// `true` if this is nothing.
+	pub fn is_zero(&self) -> bool { self.0 == 0 }
+
+	/// Everything.
+	pub fn one() -> Perquill { Perquill(1_000_000_000_000_000_000) }
+
+	/// Construct new instance where `x` is in quilltionths. Value equivalent to `x / 1,000,000,000,000,000,000`.
+	pub fn from_quilltionths(x: u64) -> Perquill { Perquill(x.min(1_000_000_000_000_000_000)) }
```

Comment:
i believe it's "quintillion"? https://www.google.com/search?q=quintillion&oq=quintil&aqs=chrome.2.69i57j0l5.4138j0j7&sourceid=chrome&ie=UTF-8

Reply:
right. horrible typo.